src/planeta/transactionHandler.js

```diff
@@ -0,0 +1,125 @@
+    const defect = inputs.filter(isOwner(address)).filter(isGoellar);
+    return defect.length > 0 ? defect[0].value : BigInt(0);
+  };
+  const getDefectInPassport = (dataBefore, dataAfter) =>
+    JSBI.subtract(bitmaskCO2Locked(dataAfter), bitmaskCO2Locked(dataBefore));
+  const isDefect = (address, dataBefore, dataAfter) =>
+    JSBI.greaterThan(
+      JSBI.add(
+        getDefectInPassport(dataBefore, dataAfter),
+        getDefectInInputs(address, inputs)
+      ),
+      BigInt(0)
+    );
+
+  // Emissions and gains functions
+  const getCO2 = (dataBefore, dataAfter) =>
+    JSBI.toNumber(
+      JSBI.subtract(bitmaskCO2(dataAfter), bitmaskCO2(dataBefore))
+    ) / 1000;
+  const getGoellars = address =>
+    // FIXME: dividing to 1e16 to get an int, the dividing by 100 to get the
+    // float...
+    JSBI.toNumber(
+      JSBI.divide(
+        JSBI.subtract(
+          BigInt(outputs.filter(isOwner(address)).filter(isGoellar)[0].value),
+          getDefectInInputs(address)
+        ),
+        BigInt("10000000000000000")
+      )
+    ) / 100;
+
+  const myPassport = BigInt(
+    inputs.filter(isMine).filter(isPassport)[0].value
+  ).toString();
+  const theirPassport = BigInt(
+    inputs.filter(isTheirs).filter(isPassport)[0].value
+  ).toString();
+  const myDataBefore = inputs.filter(isMine).filter(isPassport)[0].data;
+  const myDataAfter = outputs.filter(isMine).filter(isPassport)[0].data;
+  const theirDataBefore = inputs.filter(isTheirs).filter(isPassport)[0].data;
+  const theirDataAfter = outputs.filter(isTheirs).filter(isPassport)[0].data;
+
+  return {
+    myAddress,
+    myDefect: isDefect(myAddress, myDataBefore, myDataAfter),
+    myGoellars: getGoellars(myAddress),
+    myPassport,
+    theirAddress,
+    theirCO2: getCO2(theirDataBefore, theirDataAfter),
+    theirDefect: isDefect(theirAddress, theirDataBefore, theirDataAfter),
+    theirGoellars: getGoellars(theirAddress),
+    theirPassport
+  };
+}
+
+export default async function process(plasma, passports, tx) {
+  const myAddress = passports[0].unspent.output.address;
+  const trade4ByteSignature = plasma.eth.abi
+    .encodeFunctionSignature(
+      EarthContractData.abi.filter(({ name }) => name === "trade")[0]
+    )
+    .replace("0x", "");
+  if (tx.from.toLowerCase() !== EarthContractData.address) {
+    return;
+  }
+  const plasmaTx = Tx.fromRaw(tx.raw);
+  const outputs = plasmaTx.outputs;
+  const isHandshake =
+    plasmaTx.inputs.filter(
+      ({ msgData }) =>
+        msgData && msgData.toString("hex").startsWith(trade4ByteSignature)
+    ).length > 0;
+  if (!isHandshake) {
+    return;
+  }
+  const isMe =
+    outputs.filter(({ address }) => address === myAddress).length > 0;
+  if (!isMe) {
+    return;
+  }
+  console.log("Transaction about me", tx);
+  const inputs = await Promise.all(
+    plasmaTx.inputs.map(({ prevout: { index, hash } }) =>
+      plasma.eth
+        .getTransaction("0x" + hash.toString("hex"))
```

Comment:
Unsure but it might be that `getTransaction` fails (bad internet connection). Can we have error handling here?